ignite/engine/engine.py

```diff
@@ -808,11 +808,13 @@ def _run_once_on_dataset(self) -> float:
 
             while True:
                 self.state.batch = self.state.output = None
+                self.state.data = {}
                 try:
                     # Avoid Events.GET_BATCH_STARTED triggered twice when data iter is restarted
                     if self.last_event_name != Events.DATALOADER_STOP_ITERATION:
                         self._fire_event(Events.GET_BATCH_STARTED)
                     self.state.batch = next(self._dataloader_iter)
+                    _add_to_state_data(self.state.data, self.state.batch, "input")
```

Comment:
Why not “batch” rather than “input” ?

Reply:
This name is used if state.batch is list/tuple or non-dict and I found that "input_0", "input_1" etc is sort of better than
"batch_0", "batch_1" etc...